packages/main/src/Icon.js

```diff
@@ -137,6 +164,40 @@ class Icon extends UI5Element {
 		await fetchI18nBundle("@ui5/webcomponents");
 	}
 
+	_onfocusin(event) {
+		if (this.interactive) {
+			this.focused = true;
+		}
+	}
+
+	_onfocusout(event) {
+		this.focused = false;
+	}
+
+	_onkeydown(event) {
+		if (this.interactive && isEnter(event)) {
+			this.fireEvent("click");
+		}
+	}
+
+	_onkeyup(event) {
+		if (this.interactive && isSpace(event)) {
+			this.fireEvent("click");
+		}
+	}
+
+	_onclick(event) {
+		event.preventDefault();
```

Comment:
I think this will be a big breaking change. All HTML elements emit click always. This should be the case here as well. Just when it is not interactive, it won't have the design.